internal/runtime/runtime_test.go

```diff
@@ -566,7 +566,7 @@ func TestTopicMatches(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			actual := topicMatches(test.incomingTopic, test.pipelineTopic)
+			actual := topicMatches(test.incomingTopic, []string{test.pipelineTopic})
```

Comment:
Expand the tests struct to have multiple topics and add cases for:

1. first topic has no wild card and doesn't match exactly and second topic does match in some way
2. first topic has levels greater than that of incoming and second topic does match in some way

Also looks like I was missing a test for a single pipeline topic that has levels greater than that of incoming